Automattic/newspack-newsletters
# feat: add php sniffer for contact methods

### All Submissions:






### Changes proposed in this Pull Request:



In the effort of consolidating our data flows, we want to enforce that every contact update goes through the same route, starting at `Newspack_Newsletters_Contacts` class to make sure we have the propper logging.

However, we can't make all the internal methods private or protected.

This PHP Sniffer helps us avoid adding any new code that will bypass our intended flow. We'll need to add this sniffers to other Newspack plugins as well. - TBD how to do that

### How to test the changes in this Pull Request:

1. checkout this branch
2. run `./vendor/bin/phpcs`
3. Confirm you see some error in 2 files. (These errors are fixed in https://github.com/Automattic/newspack-newsletters/pull/1574)
4. Does the message look good?
5. Try to add a method to one of the forbidden methods anywhere in the code.
6. Make sure the sniffer catches it

PS - the linting errors in the CI job are expected until we merge https://github.com/Automattic/newspack-newsletters/pull/1574

### Other information:





---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207942575891805
